clang-tools-extra/clang-tidy/modernize/ReturnBracedInitListCheck.cpp

```diff
@@ -17,11 +17,21 @@ using namespace clang::ast_matchers;
 namespace clang::tidy::modernize {
 
 void ReturnBracedInitListCheck::registerMatchers(MatchFinder *Finder) {
-  // Skip list initialization and constructors with an initializer list.
+  auto SemanticallyDifferentContainer = allOf(
+      argumentCountIs(2), hasArgument(0, hasType(isInteger())),
```

Comment:
this may not work for ```vector( size_type count, const T& value,  const Allocator& alloc = Allocator() );```

Reply:
One more problem here, this check uses TK_IgnoreUnlessSpelledInSource, so if first argument wont be integer but some class that got conversion operator to integer, then with hasArgument you will see here actually that class.
This is a reason why I suggest to check constructor parameter instead of argument type.
Simply do: ```hasDeclaration(cxxConstructorDecl(unless(parameterCountIs(0)), hasParameter(0, hasType(...))))```

Reply:
Do you think we should ignore all when the record has constructor with std::initialList? The fix has potential risk to change program behavior.

For example, it should not change from () to {}.
```
class T {
T(int);
T(std::initialList<int>);
};

T(1);
T{1};
```

Reply:
In short yes, but we would need to somehow check a type, this mean find if there is constructor with std::initializer_list, read template type, and compare it against type we pass, or type we use. 

Simply to still find issues when types wound not match and conversion woudn't be done anyway.